provider/eureka.go

```
@@ -0,0 +1,131 @@
+package provider
+
+import (
+	"github.com/ArthurHlt/go-eureka-client/eureka"
+	log "github.com/Sirupsen/logrus"
+	"github.com/containous/traefik/safe"
+	"github.com/containous/traefik/types"
+	"io/ioutil"
+	"strconv"
+	"strings"
+	"text/template"
+	"time"
+)
+
+// Eureka holds configuration of the Eureka provider.
+type Eureka struct {
+	BaseProvider `mapstructure:",squash"`
+	Endpoint     string
+	Delay        string
+}
+
+// Provide allows the provider to provide configurations to traefik
+// using the given configuration channel.
+func (provider *Eureka) Provide(configurationChan chan<- types.ConfigMessage, pool *safe.Pool, _ []types.Constraint) error {
+
+	configuration, err := provider.buildConfiguration()
+	if err != nil {
+		log.Errorf("Failed to build configuration for Eureka, error: %s", err)
+		return nil
```

You should return the error.